### ISSUE TITLE
SD Card writing access

### ISSUE THREAD

**Issue by GammaG** [PrivacyApps/document-viewer#294]
Hey all,

I wondered is there a way to grant the application rights to edit files on the sd card? 
I have all my PDF in a folder there and it's actually pretty annoing that it's not possible to delete them from the document viewer directly. Especially since there is a delete function available, but does not work if the pdf is on the sd card.
Nice Regards

**Comment by Grossdm** [comment]
If you would be interested in a work-around for this and you have rooted your phone, you could consider using an Xposed module to redirect this app (or any other you choose) to your real sdcard.

Just a thought. 👍

**Comment by GammaG** [comment]
Yeah thanks for the advice, but I know this already. The problem is I use it mostly on a samsung tablet and samsung checks the kernel version for updates since they come as packages and replace parts. So if you root a device like this the official updates will fail. So before this thing is end of live or there is a good linageos version I won't root. 
I'm writing here exactly because of lot of applications just access the sd card once and request permission after that it works. This application on the other hand expects you to give a total path to the folder to search and won't ask for permission - reading is always allowed - so this works but all file operations fail so you cannot remove/rename/delete inside the application.

**Comment by Grossdm** [comment]
Hmm. There's a chance that we may be able to help each other here.
I also have a Samsung tablet.
It is from Verizon, a Galaxy Note 10.1 2014 Edition (SM-P605V).

A few years back, I had my tablet rooted and I still had trouble accessing the sdcard - it would lose access each time that I rebooted. I didn't understand your specific case, but I found a solution to the sdcard access on my tablet.
BTW, I acquired a Samsung phone with the tablet (package deal). The phone did not have the same access issue.

I ended up with a simple shell script that would run on each boot. It would be great if my "fix" would work for you.


I screwed up my tablet. I installed Busybox, and it (user error) overwrote some Toolbox links - `cat` is one that I know I lost.
The current situation is that I've lost root after a factory reset attempt.
I have not tried to fully rewrite the OEM ROM with Kies or whatever the software is called for that.

If you have info/knowledge about flashing Samsungs I could use some guidance. It's been a few years since I've flashed any ROMs.

How does any of this sound?

TIA,
Doug

**Comment by GammaG** [comment]
Well I flashed oneplus 2 as my latest attempt. In general I can say https://wiki.lineageos.org/devices/n1awifi the installation guides linageos provides are quite good. I experienced that you should not recover from anywhere, setting up a new system is the only thing that works well - at least did for me, had wireless drops and other bugs otherwise. 
I have Galaxy Tab A 10.1 (2016) - SMT580

**Comment by Grossdm** [comment]
Well, I don't know what TWPR is and I've only seen Magisk in a few places - I think it wouldn't run on KitKat, which is where I've been stuck. I had forgotten about that odin app. IIRC, it seemed like it had more capabilities than kies.

So Lineage has a system to flash the ROM built-in or specific to Lineage?

I certainly need the entire ROM flashed, as the reset clearly did not rewrite the partition with Toolbox on it.

Does your tablet only have Wi-Fi? Mine has the cellular modem/radio also.
I wonder about compatibility with Verizon's network and Lineage. I know almost nothing about Lineage - something like an Android fork or the AOSP?


Is there a chance I could help you with sdcard access?

Regarding root and Samsung (Verizon makes it even more difficult), I struggled with no updates for this aging phone.

This is the first smartphone that I've had that was NOT a Motorola.
I certainly would not have bought Samsung if I had known what their devices were like.

Before any of those specs for resistance to water, dirt, etc. were even discussed, I had bounced my Motorolas off concrete too many times to count.
I even let one sit in a pitcher of beer for a few minutes. No problems at all.
If like a new Moto, but I've heard that they sold off their mobile device division!
That's just crazy. I remember being a little kid, and Dad had a Motorola car phone.
Before cellular networks - it had a powerful radio to reach whatever towers they had then.

Anyway, sorry for getting side tracked there.

**Comment by Grossdm** [comment]
Oh, did you mean TWRP, BTW?
I **do** know what that is! ☺

**Comment by GammaG** [comment]
LinageOS is as it says an OS, so you can install it instead of the stock rom. Its the main alternative Android project and most of the alternatives are using it as basis. TWRP is Team Win Recovery Project just google it is quite mighty. - Ah sry I mixed up RP then. 
Well I like OnePlus since they don't lock they bootloader with a key so oem unlock is easy and requires no exploit. Also they implement project treble now in most of they devices witch is nice, that solves the driver problems in custom roms and should increase the speed for security updates. Honor and Huawei (or how you write it) both closed they bootloader now as well.
